[Board] Post presentation - Command 파트 개발

codingbaraGo opened the pull request:
<!-- 제목은 `[도메인] PR 제목`으로 작성해주세요. (ex) [User] 소셜 로그인 필터 리팩토링 -->
<!-- PR의 타입은 Label을 통해 나타내주세요. -->

### 💻 작업 내용
<!-- 진행한 작업 내용에 대해 작성해주세요. -->
- Visitor를 생성해주는 VisitorProvider 개발
- Command controller 개발

### ✨ 리뷰 포인트
<!-- (ex) query 가 너무 복잡한 것 같은데 이 위주로 봐주세요. -->
특별한것 없는 presentation 계층입니다.
컨트롤러에서 HTTP response 반환에 body를 사용하는 대신, 적절한 status와 location 헤드를 사용하였습니다.
적절하게 잘 사용되었는지 확인 부탁드립니다.

### 📝 메모
<!-- PR 관련 전달사항이 있다면 이곳에 작성해주세요. -->
VisitorProvider의 경우 Auth 도메인의 관련 코드와 함께 봐야 이해할 수 있습니다.
해당 코드 PR때 VisitorProvider 클래스에 링크 걸어두겠습니다.


### 🎯 관련 이슈
<!-- pr이 merge 되면 이슈가 자동으로 close 되도록 합니다. 만약 자동 close 를 하지 않고 이슈만 링크한다면 closed 키워드를 삭제해주세요.-->
#68

codingbaraGo committed:
refactor(post): Post aggregate에서 board name을 갖지 않도록 수정
- Board aggregate에서 boardName 수정시 정합성 문제 발생
- BoardVO가 boardName을 갖지 않도록 수정

codingbaraGo committed:
feat(auth): Visitor provider 개발 (#68)
- Http request를 보낸 클라이언트의 정보를 담는 Visitor 클래스 생성기 개발

codingbaraGo committed:
feat(post): Presentation 계층 DTO 정의 (#68)
- Command DTO 정의
- DTO Converter 개발

codingbaraGo committed:
feat(post): Post command controller 개발 (#68)

vwx12345 commented on src/main/java/com/simpleboard/board/board/presentation/util/VisitorProvider.java:
```diff
@@ -0,0 +1,92 @@
+    if (onlyGuest) return false;
+
+    // 그 외는 로그인 사용자로 간주
+    return false;
```
이부분 true로 변경해야할 것 같습니다.

vwx12345 replied:
이부분 true로 변경해야할 것 같습니다.

vwx12345 commented on src/main/java/com/simpleboard/board/board/presentation/util/VisitorProvider.java:
```diff
@@ -0,0 +1,92 @@
+   * <p>Security context holder에서 Authentication 정보를 추출
+   *
+   * <ul>
+   *   <li>Fingerprint를 추춣하여 vid 주입
```
이부분 오타가 있습니다.

vwx12345 commented on src/main/java/com/simpleboard/board/board/presentation/util/VisitorProvider.java:
```diff
@@ -0,0 +1,92 @@
+public class VisitorProvider {
+
+  private final String FINGERPRINT = "fingerprint";
+  private final String ROLE_GUEST = "ROLE_GUEST";
```
상수라면 private static final 이 낫지 않을까요?

codingbaraGo replied:
> 상수라면 private static final 이 낫지 않을까요?

어짜피 bean으로 관리되니 성능차이가 안난다 생각하고 안붙였는데,
static 을 붙이는 편이 의도가 더 명확하겠네요.
붙이겠습니다.

codingbaraGo committed:
Merge branch 'develop' of https://github.com/SimpleBoardTeam/simpleBoard-server into feat/board-post-presentation/#68

codingbaraGo committed:
refactor(post): 코드리뷰 반영
- VisitorProvider의 문자열 상수에 static 추가
- 주석 오타 수정
- VisitorProvider::isMember 메서드의 분기 반환값 수정

vwx12345 reviewed (approved):
다 반영하셨군요. 고생하셨습니다.
